src/autometrics/decorator.py

```diff
@@ -29,49 +36,103 @@ def autometrics(
     *,
     objective: Optional[Objective] = None,
 ):
+    def track_result_ok(start_time: float, function: str, module: str, caller: str):
+        get_tracker().finish(
+            start_time,
+            function=function,
+            module=module,
+            caller=caller,
+            objective=objective,
+            result=Result.OK,
+        )
+
+    def track_result_error(
+        exception: Exception,
+        start_time: float,
+        function: str,
+        module: str,
+        caller: str,
+    ):
+        get_tracker().finish(
+            start_time,
+            function=function,
+            module=module,
+            caller=caller,
+            objective=objective,
+            result=Result.ERROR,
+        )
+        # Reraise exception
+        raise exception
```

Comment:
Why should the exception be raised here? It makes more sense for the track function to only track. Also: the decorator functions that are calling this function also raise the exception.